solutions/usaco-857.mdx

```diff
@@ -15,86 +15,154 @@ functions ...
 Similar to the second solution from the analysis.
 
 <LanguageSection>
+<CPPSection>
+	
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+set<int> possible;
+
+void get_possible(
+	int day,
+	int a_tank, vector<int> a_buckets,
+	int b_tank, vector<int> b_buckets
+) {
+	// Last day, add the amount of milk in the first tank.
+	if (day == 4) {
+		possible.insert(a_tank);
+		return;
+	}
+	// This transfers every possible bucket from barn x to barn y.
+	for (int i = 0; i < a_buckets.size(); i++) {
+		// Bucket to transfer.
+		int t = a_buckets[i];
+		// Creates a new copy of available buckets, and transfers the ith one.
+		vector<int> new_a = a_buckets;
+		new_a.erase(begin(new_a) + i);
+		vector<int> new_b = b_buckets;
+		new_b.push_back(t);
+
+		// Recursively call the function with the new buckets and tank amounts.
+		get_possible(day + 1, b_tank + t, new_b, a_tank - t, new_a);
+	}
+}
+
+int main() {
+	freopen("backforth.in", "r", stdin);
+	freopen("backforth.out", "w", stdout);
+	vector<int> a(10);
+	for (int& i : a) {
+		cin >> i;
+	}
+	vector<int> b(10);
+	for (int& i : b) {
+		cin >> i;
+	}
+
+	get_possible(0, 1000, a, 1000, b);
+	cout << possible.size() << endl;
+}
+```
+</CPPSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class BackForth {
+	static HashSet<Integer> possible = new HashSet<>();
+	
+	public static void main(String[] args) throws IOException {
+		BufferedReader read = new BufferedReader(new FileReader("backforth.in"));
+		
+		StringTokenizer aST = new StringTokenizer(read.readLine());
+		StringTokenizer bST = new StringTokenizer(read.readLine());
+		List<Integer> a = new ArrayList<>();
+		List<Integer> b = new ArrayList<>();
+		for (int i = 0; i < 10; i++) {
+			a.add(Integer.parseInt(aST.nextToken()));
+			b.add(Integer.parseInt(bST.nextToken()));
+		}
+		
+		getPossible(0, 1000, a, 1000, b);
+
+		PrintWriter written = new PrintWriter("backforth.out");
+		written.println(possible.size());
+		written.close();
```

Comment:
```suggestion
		PrintWriter out= new PrintWriter("backforth.out");
		out.println(possible.size());
		out.close();
```

Reply:
happy to commit this, but could i have a reason?

Reply:
Written is kinda odd for a printwriter since it's actively writing onto a file, calling it that would be like calling a scanner "inputted"

Reply:
you could also do "write" to match the scanner if u want

Reply:
done